assets/javascripts/discourse-assign/initializers/extend-for-assigns.js.es6

```diff
@@ -112,14 +194,195 @@ function registerTopicFooterButtons(api) {
       return this.site.mobileView;
     },
     classNames: ["assign"],
-    dependentKeys: [
-      "topic.assigned_to_user",
-      "topic.assigned_to_group",
-      "currentUser.can_assign",
-      "topic.assigned_to_user.username",
-    ],
+    dependentKeys: DEPENDENT_KEYS,
     displayed() {
-      return this.currentUser && this.currentUser.can_assign;
+      return (
+        this.currentUser &&
+        this.currentUser.can_assign &&
+        !this.topic.isAssigned()
+      );
+    },
+  });
+
+  api.registerTopicFooterButton({
+    id: "unassign-mobile",
+    icon() {
+      return this.topic.isAssigned() ? "user-times" : "user-plus";
+    },
+    translatedTitle() {
+      defaultTitle(this);
+    },
+    translatedAriaLabel() {
+      defaultTitle(this);
+    },
+    translatedLabel() {
+      const user = this.get("topic.assigned_to_user");
+      const group = this.get("topic.assigned_to_group");
+      const label = I18n.t("discourse_assign.unassign.title");
+
+      if (user) {
+        return htmlSafe(
+          `<span class="unassign-label"><span class="text">${label}</span><span class="username">${
+            user.username
+          }</span></span>${renderAvatar(user, {
+            imageSize: "small",
+            ignoreTitle: true,
+          })}`
+        );
+      } else if (group) {
+        return htmlSafe(
+          `<span class="unassign-label">${label}</span> @${group.name}`
+        );
+      }
+    },
+    action() {
+      if (!this.get("currentUser.can_assign")) {
+        return;
+      }
+
+      const taskActions = getOwner(this).lookup("service:task-actions");
+      const topic = this.topic;
+
+      this.set("topic.assigned_to_user", null);
+      this.set("topic.assigned_to_group", null);
+      taskActions.unassign(topic.id).then(() => {
+        this.appEvents.trigger("post-stream:refresh", {
+          id: topic.postStream.firstPostId,
+        });
+      });
+    },
+    dropdown() {
+      return (
+        this.currentUser &&
+        this.currentUser.can_assign &&
+        this.topic.isAssigned()
+      );
+    },
+    classNames: ["assign"],
+    dependentKeys: DEPENDENT_KEYS,
+    displayed() {
+      // only display the button in the mobile view
+      return this.site.mobileView && this.topic.isAssigned();
+    },
+  });
```

Comment:
Currently there is no way to render a `topicFooterDropdown` and its content as individual items as inline buttons when in mobile. So I have registered 3 (`unassign-mobile`, `assign-self-mobile`, `reassign-mobile`) `topicFooterButtons` that will only be displayed on mobile as a workaround. Maybe @jjaffeux and I can look out building out a better solution on the `topicFooterDropdown` API side of things.